qemu/tests/invalid_cpu_device_hotplug.py

```diff
@@ -0,0 +1,129 @@
+import json
+import logging
+
+from virttest import arch
+from virttest import error_context
+from virttest.qemu_monitor import QMPCmdError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug invalid vcpu device.
+
+    1) Boot up guest without vcpu device.
+    2) Hotplug an invalid vcpu device we want
+    3) Check error messages and analyze if it is correct
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    not_match_err = ("Hotplug %s failed but the error description does not "
+                     "match: '%s'")
+    expected_info = "Hotplug %s failed as expected, error description: '%s'"
+    hotplug_pass_err = "Still able to hotplug %s via qmp"
+
+    def hotplug_inuse_vcpu():
+        """Hotplug 2 in use vcpu devices: main vcpu and duplicate vcpu"""
+        # Main vCPU
+        main_vcpu_props = {}
+        for vcpu_prop in vcpu_props:
+            main_vcpu_props.setdefault(vcpu_prop, "0")
+        vm.params["vcpu_props_main_vcpu"] = json.dumps(main_vcpu_props)
```

Comment:
What does 'main vcpu' mean?

Reply:
I think the `socket-id=0,core-id=0,thread-id=0` is the main CPU, it can not be plugged and unplugged.